bots/common/src/main/java/org/openjdk/skara/bots/common/BotUtils.java

```diff
@@ -42,15 +42,8 @@ public class BotUtils {
      */
     public static Optional<JdkVersion> getVersion(PullRequest pr) {
         String confFile;
-        try {
-            confFile = pr.repository().fileContents(".jcheck/conf", pr.headHash().hex());
-        } catch (UncheckedRestException e) {
-            if (e.getStatusCode() == 404) {
-                confFile = pr.repository().fileContents(".jcheck/conf", pr.targetRef());
-            } else {
-                throw e;
-            }
-        }
+        confFile = pr.repository().fileContents(".jcheck/conf", pr.headHash().hex())
+                .orElse(pr.repository().fileContents(".jcheck/conf", pr.targetRef()).orElseThrow());
```

Comment:
Could you add a message to the exception in orElseThrow describing what's happening. Something like
```
() -> new RuntimeException("Missing .jcheck/conf in both src and target of PR " + pr)
```

Also, no need to split declaration of confFile to a separate line anymore, nor declare the type.

Reply:
But Eriks suggested message sounds a bit contradictory. I suggest instead using:
"Could not find .jcheck/conf in neither src or target of PR"